File: VB/WindowsFormsApplication1/DragHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Utils.Mdi
Imports DevExpress.XtraTab
Imports DevExpress.XtraTab.ViewInfo
Imports DevExpress.XtraTabbedMdi

Namespace DragDataBetweenMdiTabbedForms
	Public Class DragHelper
		Private manager As XtraTabbedMdiManager

		Public Sub New(ByVal manager As XtraTabbedMdiManager)
			Me.manager = manager
		End Sub

		Public WriteOnly Property SelectNewPageOnDrag() As Boolean
			Set(ByVal value As Boolean)
				RemoveHandler MdiClient.DragOver, AddressOf OnMdiClientDragOver
				If value Then
					AddHandler MdiClient.DragOver, AddressOf OnMdiClientDragOver
				End If
			End Set
		End Property

		Private ReadOnly Property MdiClient() As MdiClient
			Get
				Return MdiClientSubclasser.GetMdiClient(manager.MdiParent)
			End Get
		End Property

		Protected Sub OnMdiClientDragOver(ByVal sender As Object, ByVal e As DragEventArgs)
			Dim xtraTab As IXtraTab = TryCast(manager, IXtraTab)
			Dim p As Point = xtraTab.ScreenPointToControl(New Point(e.X, e.Y))
			Dim hitInfo As BaseTabHitInfo = manager.CalcHitInfo(p)
			If hitInfo.HitTest = XtraTabHitTest.PageHeader Then
				manager.SelectedPage = CType(hitInfo.Page, XtraMdiTabPage)
			End If
		End Sub
	End Class
End Namespace

File: VB/WindowsFormsApplication1/DragHelper.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.Utils.Mdi;
using DevExpress.XtraTab;
using DevExpress.XtraTab.ViewInfo;
using DevExpress.XtraTabbedMdi;

namespace DragDataBetweenMdiTabbedForms
{
    public class DragHelper
    {
        XtraTabbedMdiManager manager;

        public DragHelper(XtraTabbedMdiManager manager)
        {
            this.manager = manager;
        }

        public bool SelectNewPageOnDrag
        {
            set
            {
                MdiClient.DragOver -= OnMdiClientDragOver;
                if (value)
                    MdiClient.DragOver += OnMdiClientDragOver;
            }
        }

        private MdiClient MdiClient
        {
            get
            {
                return MdiClientSubclasser.GetMdiClient(manager.MdiParent);
            }
        }

        protected void OnMdiClientDragOver(object sender, DragEventArgs e)
        {
            IXtraTab xtraTab = manager as IXtraTab;
            Point p = xtraTab.ScreenPointToControl(new Point(e.X, e.Y));
            BaseTabHitInfo hitInfo = manager.CalcHitInfo(p);
            if (hitInfo.HitTest == XtraTabHitTest.PageHeader)
                manager.SelectedPage = (XtraMdiTabPage)hitInfo.Page;
        }
    }
}
